File: DailyNews/src/main/java/com/jw/dailyNews/fragment/FragmentMe.java
package com.jw.dailyNews.fragment;

import android.app.AlertDialog;
import android.graphics.Color;
import android.os.Handler;
import android.os.Message;
import android.view.View;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.jw.dailyNews.R;
import com.jw.dailyNews.base.BaseFragment;
import com.jw.dailyNews.utils.CacheUtils;
import com.jw.dailyNews.utils.ThemeUtils;
import com.jw.dailyNews.wiget.ElasticTouchListener;
import com.jw.dailyNews.wiget.LoadingPage;

import java.util.HashMap;

import Lib.AuthManager;
import butterknife.BindView;
import butterknife.ButterKnife;
import cn.sharesdk.framework.Platform;
import cn.sharesdk.framework.ShareSDK;
import cn.sharesdk.sina.weibo.SinaWeibo;
import de.hdodenhof.circleimageview.CircleImageView;
/**
 * 创建时间：2017/6/21
 * 更新时间 2017/10/30 15:52
 * 版本：
 * 作者：Mr.jin
 * 描述：
 */

public class FragmentMe extends BaseFragment implements View.OnClickListener {

    @BindView(R.id.tv_mName)
    TextView tvMName;
    @BindView(R.id.tv_mDesc)
    TextView tvMDesc;
    public static RelativeLayout rlMe;
    @BindView(R.id.tv_mWeibo)
    TextView tvMWeibo;
    @BindView(R.id.tv_mFans)
    TextView tvMFans;
    @BindView(R.id.tv_mConcerns)
    TextView tvMConcerns;

    private CircleImageView ivMe;
    private AlertDialog dialog;
    private HashMap<String, Object> userInfos;
    private Platform Sina= ShareSDK.getPlatform(SinaWeibo.NAME);

    private Handler mHandler = new Handler(new Handler.Callback() {
        @Override
        public boolean handleMessage(Message msg) {
            tvMName.setText((String) userInfos.get("name"));
            tvMDesc.setText((String) userInfos.get("description"));
            tvMWeibo.setText(userInfos.get("credit_score") + "");
            tvMFans.setText(userInfos.get("followers_count") + "");
            tvMConcerns.setText(userInfos.get("friends_count") + "");
            Glide.with(getContext()).load(userInfos.get("profile_image_url")).into(ivMe);
            return false;
        }
    }) ;


    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.ci_me:
                if (AuthManager.getInstance().isValid(Sina))
                    showLoginDialog();
                else {
                    AuthManager.getInstance().auth(Sina,new AuthManager.AuthListener() {
                        @Override
                        public void onSuccess() {
                            updateUserInfos();
                        }
                    });
                }
                break;
            case R.id.cancel_exit:
                dialog.dismiss();
                break;
            case R.id.ok_exit:
                dialog.dismiss();
                AuthManager.getInstance().exitAutoSina();
                ivMe.setImageResource(R.drawable.icon_default_user);
                tvMName.setText("");
                tvMDesc.setText("");
                tvMWeibo.setText("");
                tvMFans.setText("");
                tvMConcerns.setText("");
                break;
        }
    }


    @Override
    protected LoadingPage.LoadResult load() {
        return LoadingPage.LoadResult.success;
    }

    @Override
    public View createSuccessView() {
        View view = View.inflate(getActivity(), R.layout.fragment_me, null);
        unbinder = ButterKnife.bind(this, view);
        ivMe = (CircleImageView) view.findViewById(R.id.ci_me);
        rlMe = (RelativeLayout) view.findViewById(R.id.rl_me);
        rlMe.setOnTouchListener(new ElasticTouchListener() {});
        ivMe.setOnClickListener(this);
        ThemeUtils.changeViewColor(rlMe, CacheUtils.getCacheInt("indicatorColor", Color.RED,getActivity()));
        updateUserInfos();
        return view;
    }

    /**
     * 更新用户资料
     */
    private void updateUserInfos() {
        if (AuthManager.getInstance().isValid(Sina))
            AuthManager.getInstance().showUser(new AuthManager.UserInfoListener() {
                @Override
                public void onSuccess(HashMap<String, Object> user) {
                    userInfos = user;
                    mHandler.sendEmptyMessage(0);
                }
            });
    }


    /**
     * 退出登录dialog
     */
    private void showLoginDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        //自定义对话框
        dialog = builder.create();
        View view = View.inflate(getContext(), R.layout.dialog_login, null);
        dialog.setView(view);
        TextView cancelExit = (TextView) view.findViewById(R.id.cancel_exit);
        TextView okExit = (TextView) view.findViewById(R.id.ok_exit);
        cancelExit.setOnClickListener(this);
        okExit.setOnClickListener(this);
        dialog.show();
    }

}

File: dailyNews/src/main/java/com/jw/dailyNews/fragment/FragmentMe.kt
package com.jw.dailyNews.fragment

import Lib.NewsManager
import android.app.AlertDialog
import android.graphics.Color
import android.os.Handler
import android.view.View
import android.widget.RelativeLayout
import android.widget.TextView
import cn.sharesdk.framework.Platform
import cn.sharesdk.framework.ShareSDK
import cn.sharesdk.sina.weibo.SinaWeibo
import com.bumptech.glide.Glide
import com.jw.dailyNews.R
import com.jw.dailyNews.base.BaseFragment
import com.jw.dailyNews.utils.CacheUtils
import com.jw.dailyNews.utils.ThemeUtils
import com.jw.dailyNews.wiget.ElasticTouchListener
import com.jw.dailyNews.wiget.LoadingPage
import de.hdodenhof.circleimageview.CircleImageView
import java.util.*

/**
 * 创建时间：2017/6/21
 * 更新时间 2017/10/30 15:52
 * 版本：
 * 作者：Mr.jin
 * 描述：
 */

class FragmentMe : BaseFragment(), View.OnClickListener, NewsManager.AuthListener, NewsManager.ShowUserListener {

    lateinit var tvMName: TextView
    lateinit var tvMDesc: TextView
    lateinit var tvMWeibo: TextView
    lateinit var tvMFans: TextView
    lateinit var tvMConcerns: TextView

    private lateinit var ivMe: CircleImageView
    private lateinit var dialog: AlertDialog
    private lateinit var userInfos: HashMap<String, Any>
    private val Sina = ShareSDK.getPlatform(SinaWeibo.NAME)

    private val mHandler = Handler(Handler.Callback {
        tvMName!!.text = userInfos!!["name"] as String
        tvMDesc!!.text = userInfos!!["description"] as String
        tvMWeibo!!.text = userInfos!!["credit_score"].toString() + ""
        tvMFans!!.text = userInfos!!["followers_count"].toString() + ""
        tvMConcerns!!.text = userInfos!!["friends_count"].toString() + ""
        Glide.with(context).load(userInfos!!["profile_image_url"]).into(ivMe!!)
        false
    })


    override fun onClick(v: View) {
        when (v.id) {
            R.id.civMef -> if (NewsManager.Companion.get().isValid(Sina))
                showExitDialog()
            else {
                NewsManager.Companion.get().auth(Sina, this)
            }
            R.id.tvCancelExit -> dialog!!.dismiss()
            R.id.tvOkExit -> {
                dialog!!.dismiss()
                NewsManager.Companion.get().exitAuth(Sina)
                ivMe!!.setImageResource(R.drawable.ic_default_user)
                tvMName!!.text = ""
                tvMDesc!!.text = ""
                tvMWeibo!!.text = ""
                tvMFans!!.text = ""
                tvMConcerns!!.text = ""
            }
        }
    }


    override fun load(): LoadingPage.LoadResult {
        return LoadingPage.LoadResult.success
    }

    override fun createSuccessView(): View {
        val view = View.inflate(activity, R.layout.fragment_me, null)
        ivMe = view.findViewById(R.id.civMef)
        rlMe = view.findViewById(R.id.rlMef)
        rlMe.setOnTouchListener(object : ElasticTouchListener() {})
        ivMe!!.setOnClickListener(this)
        ThemeUtils.changeViewColor(rlMe, CacheUtils.getCacheInt("indicatorColor", Color.RED))
        updateUserInfos()
        return view
    }

    /**
     * 更新用户资料
     */
    private fun updateUserInfos() {
        if (NewsManager.Companion.get().isValid(Sina)) {
            //ThemeUtils.show(getActivity(),"正在获取用户信息");
            NewsManager.Companion.get().showUser(Sina, this)
        }
    }


    /**
     * 退出登录dialog
     */
    private fun showExitDialog() {
        val builder = AlertDialog.Builder(context)
        //自定义对话框
        dialog = builder.create()
        val view = View.inflate(context, R.layout.dialog_login, null)
        dialog!!.setView(view)
        val cancelExit = view.findViewById<TextView>(R.id.tvCancelExit)
        val okExit = view.findViewById<TextView>(R.id.tvOkExit)
        cancelExit.setOnClickListener(this)
        okExit.setOnClickListener(this)
        dialog!!.show()
    }

    override fun onAuthSuccess(platform: Platform) {
        //ThemeUtils.show(getActivity(),"授权成功");
        updateUserInfos()
    }

    override fun onShowUserSuccess(userInfos: HashMap<String, Any>) {
        this.userInfos = userInfos
        //ThemeUtils.show(getActivity(),"获取用户信息成功");
        mHandler.sendEmptyMessage(0)
    }

    companion object {
        lateinit var rlMe: RelativeLayout
    }
}
